modules/partition-distribution/src/main/java/org/apache/ignite/internal/partitiondistribution/AssignmentsChain.java

```diff
@@ -64,31 +71,70 @@ public AssignmentsChain replaceLast(Assignments newLast) {
      * @param newLast New last link.
      * @return new AssignmentsChain.
      */
-    public AssignmentsChain addLast(Assignments newLast) {
+    public AssignmentsChain addLast(Assignments newLast, long configurationTerm, long configurationIndex) {
         assert !chain.isEmpty() : "Assignments chain is empty.";
 
-        List<Assignments> newChain = new ArrayList<>(chain);
+        List<AssignmentsLink> newChain = new ArrayList<>(chain);
 
-        newChain.add(newLast);
+        newChain.add(new AssignmentsLink(newLast, configurationTerm, configurationIndex));
 
         return new AssignmentsChain(newChain);
     }
 
+
+    /**
+     * Gets the next link in the chain after the given link.
+     *
+     * @param link The link to get the next one from.
+     * @return The next link in the chain, or {@code null} if the given link is the last one in the chain.
+     */
+    public @Nullable AssignmentsLink nextLink(AssignmentsLink link) {
```

Comment:
I didn't expect such method to be implemented. I'd rather add AssignmentsLink.next().

Reply:
Agreed to revert it to initial one.